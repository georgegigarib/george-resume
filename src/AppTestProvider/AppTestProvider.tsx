import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { vi } from 'vitest'

vi.mock('swiper/react', () => ({
  Swiper: ({
    children,
    onSlideChange,
  }: {
    children: React.ReactNode
    onSlideChange?: (params: { realIndex: number }) => void
  }) => {
    onSlideChange?.({ realIndex: 0 })
    return <div data-testid="swiper">{children}</div>
  },
  SwiperSlide: ({ children }: { children: React.ReactNode }) => <div data-testid="swiper-slide">{children}</div>,
}))

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
