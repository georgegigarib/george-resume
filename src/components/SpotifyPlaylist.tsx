export default function SpotifyPlaylist() {

    return (
      <div className="">
        aqui dejo algo
        <iframe className="rounded-xl" src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator" width="100%" height="352" frameBorder="0" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy"></iframe>
      </div>
    );
  };
  
  