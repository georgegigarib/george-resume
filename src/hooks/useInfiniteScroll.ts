import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
