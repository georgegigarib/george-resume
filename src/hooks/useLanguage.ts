import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
