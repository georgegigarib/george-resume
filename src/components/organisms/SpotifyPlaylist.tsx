import React from 'react';
import { useIsMobile } from '@/hooks/useIsMobile';
import { useTranslation } from 'react-i18next';

export default function SpotifyPlaylist(): React.ReactElement {
  const isMobile = useIsMobile();
  const { t } = useTranslation()
  
  return (
    <div className="flex flex-col justify-between h-full p-4  rounded-xl bg-gray-200 dark:bg-gray-800">
      <div className="mb-4">
        <h2 className="text-xl font-bold mb-2 text-gray-700 dark:text-gray-300">
        {t('music.favoritePlaylist.title')}
        </h2>
        <p className="text-gray-700 dark:text-gray-300">
          {t('music.favoritePlaylist.description')}
        </p>
      </div>
      
      <div className="mt-auto">
        <iframe
          className="rounded-xl"
          src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator"
          width='100%'
          height={isMobile ? "152px" : "379px"}
          allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
          loading="lazy"
        ></iframe>
      </div>
    </div>
  );
}
