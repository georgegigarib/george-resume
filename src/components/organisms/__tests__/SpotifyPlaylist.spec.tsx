import { render, screen } from '@testing-library/react'
import { describe, it, expect, vi, Mock } from 'vitest'
import { useIsMobile } from '@/hooks/useIsMobile'
import SpotifyPlaylist from '../SpotifyPlaylist'
import AppTestProvider from '@/AppTestProvider/AppTestProvider'

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn()
}))

describe('SpotifyPlaylist Component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <SpotifyPlaylist />
      </AppTestProvider>
    )

  it('renders correctly in mobile view', () => {
    ;(useIsMobile as Mock).mockReturnValue(true)

    renderComponent()

    expect(screen.getByText('My Favorite Playlist')).toBeVisible()
    expect(
      screen.getByText(
        'Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!'
      )
    ).toBeVisible()

    const iframe = screen.getByTestId('spotify-playlist-iframe')
    expect(iframe).toHaveAttribute('height', '152px')
  })

  it('renders correctly in desktop view', () => {
    ;(useIsMobile as Mock).mockReturnValue(false)

    renderComponent()

    expect(screen.getByText('My Favorite Playlist')).toBeVisible()
    expect(
      screen.getByText(
        'Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!'
      )
    ).toBeVisible()

    const iframe = screen.getByTestId('spotify-playlist-iframe')
    expect(iframe).toHaveAttribute('height', '379px')
  })
})
