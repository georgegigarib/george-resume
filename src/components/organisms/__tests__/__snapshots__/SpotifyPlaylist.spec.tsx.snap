// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpotifyPlaylist Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-between h-full p-4  rounded-xl 
    bg-app-lightModeBoxes-box16 
    dark:bg-app-darkModeBoxes-box16 
    dark:border-app-lightModeBoxes-box16 
    border-app-darkModeBoxes-box16
    border-[2px]"
  >
    <div
      class="mb-4"
    >
      <h2
        class="text-xl font-bold mb-2 text-black dark:text-white"
      >
        My Favorite Playlist
      </h2>
      <p
        class="text-black dark:text-white"
      >
        Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!
      </p>
    </div>
    <div
      class="mt-auto"
    >
      <iframe
        allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
        class="rounded-xl"
        data-testid="spotify-playlist-iframe"
        height="379px"
        loading="lazy"
        src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SpotifyPlaylist Component > renders correctly in desktop view 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-between h-full p-4  rounded-xl 
    bg-app-lightModeBoxes-box16 
    dark:bg-app-darkModeBoxes-box16 
    dark:border-app-lightModeBoxes-box16 
    border-app-darkModeBoxes-box16
    border-[2px]"
  >
    <div
      class="mb-4"
    >
      <h2
        class="text-xl font-bold mb-2 text-black dark:text-white"
      >
        My Favorite Playlist
      </h2>
      <p
        class="text-black dark:text-white"
      >
        Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!
      </p>
    </div>
    <div
      class="mt-auto"
    >
      <iframe
        allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
        class="rounded-xl"
        data-testid="spotify-playlist-iframe"
        height="379px"
        loading="lazy"
        src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SpotifyPlaylist Component > renders correctly in mobile view 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-between h-full p-4  rounded-xl 
    bg-app-lightModeBoxes-box16 
    dark:bg-app-darkModeBoxes-box16 
    dark:border-app-lightModeBoxes-box16 
    border-app-darkModeBoxes-box16
    border-[2px]"
  >
    <div
      class="mb-4"
    >
      <h2
        class="text-xl font-bold mb-2 text-black dark:text-white"
      >
        My Favorite Playlist
      </h2>
      <p
        class="text-black dark:text-white"
      >
        Enjoy my personal selection of songs in this playlist. Listen to it and let yourself be carried away by the rhythm!
      </p>
    </div>
    <div
      class="mt-auto"
    >
      <iframe
        allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture"
        class="rounded-xl"
        data-testid="spotify-playlist-iframe"
        height="152px"
        loading="lazy"
        src="https://open.spotify.com/embed/playlist/2Z8vIZcBXeeCfcmkffrTlR?utm_source=generator"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;
